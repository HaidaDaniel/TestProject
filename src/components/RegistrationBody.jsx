/** @format */

import { Row } from 'react-bootstrap'

import {
  StyledForm,
  StyledCol,
  StyledHeading,
  StyledFormGroup,
  StyledFormLabel,
  StyledFormControl,
  StyledSubmitButton,
} from '../styled/RegistrationBodyStyles'

function RegistrationBody() {
  return (
    <Row>
      <StyledForm>
        <StyledCol xs={12} md={6}>
          <StyledHeading>Registration Form</StyledHeading>
          <StyledFormGroup>
            <StyledFormLabel>Email address</StyledFormLabel>
            <StyledFormControl type='email' placeholder='Email address' />
          </StyledFormGroup>
          <StyledFormGroup>
            <StyledFormLabel>First name</StyledFormLabel>
            <StyledFormControl placeholder='First name' />
          </StyledFormGroup>
          <StyledFormGroup>
            <StyledFormLabel>LastName</StyledFormLabel>
            <StyledFormControl placeholder='LastName' />
          </StyledFormGroup>
          <StyledFormGroup>
            <StyledFormLabel>Password</StyledFormLabel>
            <StyledFormControl type='password' placeholder='Password' />
          </StyledFormGroup>
          <StyledSubmitButton variant='primary' type='submit'>
            Registration
          </StyledSubmitButton>
        </StyledCol>
      </StyledForm>
    </Row>
  )
}

export default RegistrationBody
