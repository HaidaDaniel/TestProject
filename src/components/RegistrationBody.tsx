/** @format */
import { useState, FC } from 'react'
import { Row, Modal, Button } from 'react-bootstrap'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import * as Yup from 'yup'
import { useNavigate } from 'react-router-dom'

import {
  StyledCol,
  StyledHeading,
  StyledFormContainer,
  StyledFormGroup,
  StyledFormLabel,
  StyledFormControl,
} from '../styled/RegistrationBodyStyles'

const RegistrationBody: FC = () => {
  const [isSuccess, setIsSuccess] = useState(false)
  const navigate = useNavigate()

  const validationSchema = Yup.object().shape({
    email: Yup.string()
      .email('Please enter a valid email address')
      .required('This field is required'),
    firstName: Yup.string().required('This field is required'),
    lastName: Yup.string().required('This field is required'),
    password: Yup.string()
      .min(8, 'Password must be at least 8 characters long')
      .required('This field is required'),
    confirmPassword: Yup.string()
      .oneOf([Yup.ref('password'), undefined], 'Passwords must match')
      .required('This field is required'),
  })

  const handleSubmit = async (values: any, { resetForm }: any) => {
    try {
      setTimeout(() => {
        console.log(values)
        setIsSuccess(true)
        resetForm()
      }, 500)
    } catch (error) {
      console.error(error)
    }
  }

  const handleCloseModal = () => {
    setIsSuccess(false)
    setTimeout(() => {
      navigate('/login')
    }, 500)
  }

  return (
    <Row>
      <StyledFormContainer>
        <StyledCol xs={12} md={6}>
          <StyledHeading>Registration Form</StyledHeading>
          <Formik
            initialValues={{
              email: '',
              firstName: '',
              lastName: '',
              password: '',
              confirmPassword: '',
            }}
            validationSchema={validationSchema}
            onSubmit={handleSubmit}>
            {({ isValid }: any) => (
              <Form>
                <StyledFormGroup>
                  <StyledFormLabel>Email address</StyledFormLabel>
                  <Field
                    name='email'
                    as={StyledFormControl}
                    type='email'
                    placeholder='Email address'
                  />
                  <ErrorMessage name='email' component='div' />
                </StyledFormGroup>
                <StyledFormGroup>
                  <StyledFormLabel>First name</StyledFormLabel>
                  <Field
                    name='firstName'
                    as={StyledFormControl}
                    placeholder='First name'
                  />
                  <ErrorMessage name='firstName' component='div' />
                </StyledFormGroup>
                <StyledFormGroup>
                  <StyledFormLabel>Last name</StyledFormLabel>
                  <Field
                    name='lastName'
                    as={StyledFormControl}
                    placeholder='Last name'
                  />
                  <ErrorMessage name='lastName' component='div' />
                </StyledFormGroup>
                <StyledFormGroup>
                  <StyledFormLabel>Password</StyledFormLabel>
                  <Field
                    name='password'
                    as={StyledFormControl}
                    type='password'
                    placeholder='Password'
                  />
                  <ErrorMessage name='password' component='div' />
                </StyledFormGroup>
                <StyledFormGroup>
                  <StyledFormLabel>Confirm Password:</StyledFormLabel>
                  <Field
                    name='confirmPassword'
                    as={StyledFormControl}
                    type='password'
                    placeholder='Confirm Password'
                  />
                  <ErrorMessage name='confirmPassword' component='div' />
                </StyledFormGroup>
                <Button type='submit' disabled={!isValid}>
                  Registration
                </Button>
              </Form>
            )}
          </Formik>
          <Modal show={isSuccess} onHide={handleCloseModal}>
            <Modal.Header closeButton>
              <Modal.Title>Success</Modal.Title>
            </Modal.Header>
            <Modal.Body>Your registration was successful.</Modal.Body>
            <Modal.Footer>
              <Button variant='primary' onClick={handleCloseModal}>
                Close
              </Button>
            </Modal.Footer>
          </Modal>
        </StyledCol>
      </StyledFormContainer>
    </Row>
  )
}

export default RegistrationBody
