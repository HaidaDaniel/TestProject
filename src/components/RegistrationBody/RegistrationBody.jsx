/** @format */

import { Form, Button, Col } from 'react-bootstrap'

function RegistrationBody() {
  return (
    <div>
      <Form className=' pt-3'>
        <Col xs={12} md={6} className='mx-auto border rounded'>
          <h3>Registration Form</h3>
          <Form.Group className='mb-3 mx-4 mt-3' controlId='formGroupEmail'>
            <Form.Label>Email address </Form.Label>
            <Form.Control type='email' placeholder='Email address' />
          </Form.Group>
          <Form.Group className='mb-3 mx-4' controlId='formGroupFirstName'>
            <Form.Label>First name </Form.Label>
            <Form.Control placeholder='First name' />
          </Form.Group>
          <Form.Group className='mb-3 mx-4' controlId='formGroupLastName'>
            <Form.Label>LastName </Form.Label>
            <Form.Control placeholder='LastName' />
          </Form.Group>
          <Form.Group className='mb-3 mx-4' controlId='formGroupPassword'>
            <Form.Label>Password</Form.Label>
            <Form.Control type='password' placeholder='Password' />
          </Form.Group>
          <Button variant='primary' type='submit' className='mb-3'>
            Registration
          </Button>
        </Col>
      </Form>
    </div>
  )
}
export default RegistrationBody
