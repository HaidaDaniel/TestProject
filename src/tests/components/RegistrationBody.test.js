import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
<<<<<<< HEAD
<<<<<<< HEAD

import RegistrationBody from '../../components/RegistrationBody';

import wrapComponentInAllProviders from '../helpers/wrapComponentInAllProviders';

describe('RegistrationBody Component', () => {
    it('should handle form submission and success modal', async () => {
=======
import RegistrationBody from '../../components/RegistrationBody';
import wrapComponentInAllProviders from '../helpers/wrapComponentInAllProviders';

describe('RegistrationBody Component', () => {
    it('should handles form submission and success modal', async () => {
>>>>>>> origin/newTest
=======
import RegistrationBody from '../../components/RegistrationBody';
import wrapComponentInAllProviders from '../helpers/wrapComponentInAllProviders';

describe('RegistrationBody Component', () => {
    it('should handles form submission and success modal', async () => {
>>>>>>> cdcb47e (1st refactor of tests description)
        render(
            wrapComponentInAllProviders(<RegistrationBody />)
        );

        fireEvent.change(screen.getByPlaceholderText('Email address'), {
            target: { value: 'test@example.com' },
        });
        fireEvent.change(screen.getByPlaceholderText('First name'), {
            target: { value: 'John' },
        });
        fireEvent.change(screen.getByPlaceholderText('Last name'), {
            target: { value: 'Doe' },
        });
        fireEvent.change(screen.getByPlaceholderText('Password'), {
            target: { value: 'password123' },
        });
        fireEvent.change(screen.getByPlaceholderText('Confirm Password'), {
            target: { value: 'password123' },
        });

        fireEvent.click(screen.getByText('Registration'));

        await waitFor(() => {
            expect(screen.getByText('Your registration was successful.')).toBeInTheDocument();
        });

        fireEvent.click(screen.getByText('Close'));

        await waitFor(() => {
            expect(screen.queryByText('Your registration was successful.')).not.toBeInTheDocument();
        });
        expect(document.body).toMatchSnapshot();
    });
});
