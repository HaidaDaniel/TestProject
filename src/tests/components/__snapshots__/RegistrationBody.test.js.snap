// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
<<<<<<<< HEAD:src/tests/__snapshots__/components/RegistrationBody.snap.js
exports[`RegistrationBody Component should handle form submission and success modal 1`] = `
========
exports[`RegistrationBody Component should handles form submission and success modal 1`] = `
>>>>>>>> origin/newTest:src/tests/components/__snapshots__/RegistrationBody.test.js.snap
=======
exports[`RegistrationBody Component should handles form submission and success modal 1`] = `
>>>>>>> cdcb47e (1st refactor of tests description)
<body
  class=""
  style="padding-right: 1024px;"
>
  <div>
    <div
      class="row"
    >
      <div
        class="sc-bcPKhP jVRLJQ"
      >
        <div
          class="sc-grXZZQ fyIqj col-md-6 col-12"
        >
          <h3
            class="sc-dkjKgF hKyBdL"
          >
            Registration Form
          </h3>
          <form
            action="#"
          >
            <div
              class="sc-hLclGa eOwcXS"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Email address
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="email"
                placeholder="Email address"
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-hLclGa eOwcXS"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                First name
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="firstName"
                placeholder="First name"
                value=""
              />
            </div>
            <div
              class="sc-hLclGa eOwcXS"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Last name
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="lastName"
                placeholder="Last name"
                value=""
              />
            </div>
            <div
              class="sc-hLclGa eOwcXS"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Password
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              class="sc-hLclGa eOwcXS"
            >
              <label
                class="sc-eDnVMP gwrerJ form-label"
              >
                Confirm Password:
              </label>
              <input
                class="sc-jSwlEQ gvQgFa form-control"
                name="confirmPassword"
                placeholder="Confirm Password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Registration
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</body>
`;
